src/metaxy/metadata_store/base.py

```diff
@@ -7,12 +7,17 @@
 from collections.abc import Iterator, Mapping, Sequence
 from contextlib import contextmanager
 from datetime import datetime, timezone
-from typing import TYPE_CHECKING, Any, Literal, overload
+from typing import Any, Literal, overload
 
 import narwhals as nw
 import polars as pl
 from typing_extensions import Self
 
+try:  # pyarrow is an optional dependency needed only for Arrow streaming paths
```

Comment:
can this be moved from the top-level imports to the specific part of the code that's actually using it?

Reply:
only if we add pyarrow to default deps. should we do that? otherwise we would need this fallback

Reply:
see the modification only imported in the specific code path that needs it (fallback stores with diff resolution)